qa-pipelines/tasks/cf-upgrade.sh

```diff
@@ -98,7 +98,48 @@ wait_for_namespace() {
 
 PROVISIONER=$(kubectl get storageclasses persistent -o "jsonpath={.provisioner}")
 
-# Deploy UAA
+# monitor_url takes a URL argument and a path to a log file
+# This will time out after 30 hours. Until then, repeatedly curl the URL with a 1-second wait period, and log the response
+# If the application state changes, print this to stdout as well
+monitor_url() {
+  local app_url=$1
+  local log_file=$2
+  local count=0
+  local last_state=
+  local new_state=
+  echo "monitoring URL ${app_url}"
+  while true; do
+    new_state=$({ curl -sI "${1}" || echo "URL could not be reached"; } | head -n1 | tee -a "${log_file}")
```

Comment:
we can call "Downtime" instead of "URL could not be reached". The summary will become more aligned for this "Downtime" test. (suggestion)

Reply:
There are various kinds of downtime though.. even if the router is up, if the app is still down you'll see a 404.. Since there's not just one kind of downtime, we want this output to inform the user that the app was actually *unreachable* (so there was no http response status), as opposed to various other types of responses that would *also* indicate downtime.